### ISSUE TITLE
Monkey patch for older GPU

### ISSUE THREAD

**Issue by S-Mahoney** [ByteDance-Seed/Depth-Anything-3#11]
Hi, big fan of the work.

Just wanted to show a workaround patch for those who also don't have a GPU that can utilise Xformers.

Following solution tested on system:

- torch 2.4.1 + cu121
- GTX 1060 (Pascal)
- Windows
- No CUDA SparseLT
- No FlashAttention compatible kernels
- No xformers binary available for this setup

With DA3-Small and DA3-Base.

```
import glob, os, torch

# Inject missing numpy.math before DA3 imports
import sys
import types
import math

# Create a fake module so "from numpy import math" works
numpy_math = types.ModuleType("numpy.math")
numpy_math = math
sys.modules["numpy.math"] = numpy_math

print("[AUTO-PATCH] Injected numpy.math shim")

# PATCH DA3 TO DISABLE XFORMERS
import sys, types, torch.nn as nn

fake_xformers = types.ModuleType("xformers")
fake_ops = types.ModuleType("xformers.ops")

# Dummy SwiGLU class
class FakeSwiGLU(nn.Module):
    def __init__(self, *a, **k):
        super().__init__()
    def forward(self, x):
        return x  # identity fallback

# Provide dummy classes/functions to avoid import errors
fake_ops.SwiGLU = FakeSwiGLU
fake_ops.MemoryEfficientAttention = FakeSwiGLU
fake_ops.flash_attention = lambda *a, **k: None
fake_ops.unbind = lambda *a, **k: None

fake_xformers.ops = fake_ops
sys.modules["xformers"] = fake_xformers
sys.modules["xformers.ops"] = fake_ops

print("[AUTO-PATCH] xformers replaced with FakeSwiGLU")

# Now we can import depth anything 3
from depth_anything_3.api import DepthAnything3
import matplotlib.pyplot as plt

device = torch.device("cuda")
model = DepthAnything3.from_pretrained("depth-anything/DA3-Base")
model = model.to(device=device)
example_path = os.path.join(os.path.dirname(__file__), "..", "assets", "examples", "SOH")
example_path = os.path.abspath(example_path)

print(f"Searching images in {example_path}")
images = sorted(glob.glob(os.path.join(example_path, "*.png")))
print(images)

prediction = model.inference(
    images,
)
# prediction.processed_images : [N, H, W, 3] uint8   array
print(prediction.processed_images.shape)
# prediction.depth            : [N, H, W]    float32 array
print(prediction.depth.shape)  
# prediction.conf             : [N, H, W]    float32 array
print(prediction.conf.shape)  
# prediction.extrinsics       : [N, 3, 4]    float32 array # opencv w2c or colmap format
print(prediction.extrinsics.shape)
# prediction.intrinsics       : [N, 3, 3]    float32 array
print(prediction.intrinsics.shape)

# Access depth maps
if hasattr(prediction, 'depth'):
    depth_maps = prediction.depth  # shape: (2, H, W)

for img in depth_maps:
    plt.imshow(img, cmap='plasma')
    plt.show()
```

This code monkey-patches numpy and xformers to allow it to run on older systems.

**Comment by haotongl** [comment]
Thank you for your generous support. I have added it to our readme doc.

**Comment by mhyatt000** [comment]
@S-Mahoney what is the FPS on GTX 1060 ??

**Comment by S-Mahoney** [comment]
@mhyatt000 

## Tested on:
assets/examples/robot_unitree.mp4 
frame width: 1024
frame height: 576

Processing:
Batch size: 8
_So we dont load the whole vid into vram_

## Results:

- Using "depth-anything/da3metric-large":
[STATS] Depth inference on video: 174 frames in 161.298s → 1.08 FPS

- Using "depth-anything/DA3-Base":
[STATS] Depth inference on video: 174 frames in 84.037s → 2.07 FPS

- Using "depth-anything/DA3-Small":
[STATS] Depth inference on video: 174 frames in 40.259s → 4.32 FPS

**Comment by ocetintas** [comment]
Have a similar issue. Just to understand the solution more clearly, if we use this solution we cannot use any models utilizing SwiGLU right? Which models could be used with a solution like this safely? I am assuming Giant models cannot be used as they return the log: [INFO ] using SwiGLU layer as FFN @S-Mahoney @haotongl
